fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarRangePredicateExtractor.java

```diff
@@ -119,9 +126,7 @@ private ScalarOperator rewrite(ScalarOperator predicate, boolean onlyExtractColu
             if (!checkStatisticsEstimateValid(extractExpr)) {
                 extractExpr.setNotEvalEstimate(true);
             }
-            // TODO: merge `setFromPredicateRangeDerive` into `setRedundant`
-            result.forEach(f -> f.setRedundant(true));
-            extractExpr.setRedundant(true);
+            result.forEach(f -> optimizerContext.addPushdownDerivedPredicate(f));
```

Comment:
`setRedundant ` is used not only for mv rewrite, is this change safe?

Reply:
setRedundant seems only used for mv rewrite?